src/Features/WcPayPromotion/DataSourcePoller.php

```diff
@@ -0,0 +1,131 @@
+<?php
+/**
+ * Handles polling and storage of specs
+ */
+
+namespace Automattic\WooCommerce\Admin\Features\WcPayPromotion;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Specs data source poller class.
+ * This handles polling specs from JSON endpoints.
+ */
+class DataSourcePoller {
+	/**
+	 * Name of data sources filter.
+	 */
+	const FILTER_NAME = 'woocommerce_admin_payment_method_promotions_data_sources';
+
+	/**
+	 * Default data sources array.
+	 */
+	const DATA_SOURCES = array(
+		'https://woocommerce.com/wp-json/wccom/marketplace-suggestions/1.0/payment-method-promotions.json',
+	);
+
+	/**
+	 * The logger instance.
+	 *
+	 * @var WC_Logger|null
+	 */
+	protected static $logger = null;
+
+	/**
+	 * Get the logger instance.
+	 *
+	 * @return WC_Logger
+	 */
+	private static function get_logger() {
+		if ( is_null( self::$logger ) ) {
+			self::$logger = wc_get_logger();
+		}
+
+		return self::$logger;
+	}
+
+	/**
+	 * Reads the data sources for specs and persists those specs.
+	 *
+	 * @return bool Whether any specs were read.
+	 */
+	public static function read_specs_from_data_sources() {
```

Comment:
Out of curiosity, will default fallback specs be added if polling fails?

Reply:
No, it will not be displayed in that case.
@pmcpinto do you think a fallback for the WooCommerce Payment method promotion would be needed?

Reply:
@pmcpinto If you want to add fallbacks for this just in case, let us know and we can follow up on this.